types/ibm-openapi-validator/index.d.ts

```diff
@@ -0,0 +1,28 @@
+// Type definitions for ibm-openapi-validator 0.15
+// Project: https://github.com/IBM/openapi-validator#readme
+// Definitions by: Rifa Achrinza <https://github.com/DefinitelyTyped>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+export interface validatorResultItem {
+    path: string;
+    message: string;
+}
+
+export interface validatorResult {
+    errors: [] | [ validatorResultItem ];
+    warnings: [] | [ validatorResultItem ];
+}
+
+/**
+ * @default false
+ */
+export type validatorParameterDefaultMode = boolean;
+
+/**
+ * OpenAPI document validator
+ * @param openApiDoc - OpenAPI document object
+ * @param defaultMode - If set to true, the validator will ignore the .validaterc file and will use the [configuration defaults](https://github.com/IBM/openapi-validator#default-values).
+ * @returns Validation results
+ */
+export function validator(openApiDoc: {[property: string]: any}, defaultMode?: validatorParameterDefaultMode): Promise<validatorResult>;
```

Comment:
Instead of defining export function validator you need to use `export = validator` Since thats how library represents it as per https://github.com/IBM/openapi-validator#node-module
Refer to http://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html

Reply:
Thanks for the references; the changes have been made.